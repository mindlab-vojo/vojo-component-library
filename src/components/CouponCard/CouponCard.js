import React from 'react'
import PropTypes from 'prop-types'

import * as CouponCard from './CouponCard.style'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import Typography from '../Typography'
import IconManager from '../IconManager'

const CouponCardComponent = ({
  children,
  id,
  isActive,
  justifyContent,
  date,
  title,
}) => {

  const renderComponent = (theme) => (
    <CouponCard.Container key={id} backgroundColor={isActive ? '#E9E5FF' : '#DDDDDD'}>
      <CouponCard.Icon>
        <IconManager
          height={'36px'}
          fill={isActive ? theme.colors.primaryColor : theme.colors.darkGreyColor}
          icon={'Coupon'}
          width={'36px'} />
      </CouponCard.Icon>
      <CouponCard.DataBlock justifyContent={justifyContent}>
        <CouponCard.Title>
          <Typography
            tag={'h4'}
            color={isActive ? theme.colors.primaryColor : theme.colors.darkGreyColor}
            type={'subtitle'}>
            {title}
          </Typography>
        </CouponCard.Title>
        {date && <CouponCard.Subtitle>
          <Typography
            type={'label'}
            color={theme.colors.lightGreyColor}>
            {`${date}`}
          </Typography>
        </CouponCard.Subtitle>}
        {children}
      </CouponCard.DataBlock>
      <CouponCard.Status>
        {isActive ?
          (<IconManager
            height={'18px'}
            width={'18px'}
            icon={'CheckOutlined'}
            fill={isActive ? (theme.colors.primaryColor) : (theme.colors.darkGreyColor)} />) :
          (<IconManager
            height={'20px'}
            width={'20px'}
            icon={'CancelOutlined'}
            fill={isActive ? theme.colors.primaryColor : theme.colors.darkGreyColor} />)}
      </CouponCard.Status>
    </CouponCard.Container>
  )

  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

CouponCardComponent.propTypes = {
  children: PropTypes.node,
  id: PropTypes.string,
  isActive: PropTypes.bool,
  justifyContent: PropTypes.string,
  date: PropTypes.string,
  title: PropTypes.string,
}

export default CouponCardComponent
