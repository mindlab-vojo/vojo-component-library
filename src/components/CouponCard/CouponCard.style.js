import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${props => props.backgroundColor};
  align-items: center;
  display: flex;
  justify-content: space-between;
  width: 100%;
  border: 2px dashed #391DDD;
  box-sizing: border-box;
  border-radius: 4px;
  padding: 16px 18px;
`

export const Icon = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
`

export const DataBlock = styled.div`
  display: flex;
  align-items: flex-start;
  width: 80%;
  justify-content: ${props => props.justifyText};
  flex-direction: column;
  margin: 0 12px;
  h3 {
      font-size: 14px;
      font-family: 'Montserrat', sans-serif;
      color: red;
      letter-spacing: 0.2px;
  }
`

export const Title = styled.div`
  margin-bottom: 4px;
`

export const Subtitle = styled.div`
  margin-bottom: 4px; 
`
export const Status = styled.div`
`