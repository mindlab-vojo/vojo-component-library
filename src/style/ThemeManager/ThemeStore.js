import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore