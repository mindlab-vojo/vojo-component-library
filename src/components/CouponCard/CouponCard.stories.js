import React from 'react';
import { withKnobs, text, boolean, color } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';

import CouponCard from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const couponCard = () => {
  const backgroundColor = color('backgroundColor', "#E9E5FF")
  const id = text('id', 'card-id')
  const justifyContent = text('justifyContent', 'space-between')
  const date = text('date', 'Descubra sites de empresas que te ajudam a gerar renda ')
  const title = text('title', 'CAMPINAS123')
  const isActive = boolean('isValid', true)

  return (
    <CouponCard
      backgroundColor={backgroundColor}
      id={id}
      isActive={isActive}
      justifyContent={justifyContent}
      date={date}
      title={title}
    />
  )
}

export default {
  title: categories.Cards,
  name: 'CouponCard',
  component: couponCard,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{ story() }</div>,
  ],
}
  