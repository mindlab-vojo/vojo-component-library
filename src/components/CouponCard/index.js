import CouponCard from './CouponCard'
export default CouponCard