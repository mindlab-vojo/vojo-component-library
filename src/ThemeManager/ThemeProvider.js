import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer