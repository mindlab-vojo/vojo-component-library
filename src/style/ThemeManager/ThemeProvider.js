import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer