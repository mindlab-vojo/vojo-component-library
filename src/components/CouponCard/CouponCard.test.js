import React from 'react';
import { render } from '@testing-library/react';

import CouponCard from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test CouponCard component render', () => {
  const component = render(
    <ThemeProvider>
      <CouponCard/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
test('Test Reverse CouponCard component render', () => {
  const component = render(
    <ThemeProvider>
      <CouponCard isReverse/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
