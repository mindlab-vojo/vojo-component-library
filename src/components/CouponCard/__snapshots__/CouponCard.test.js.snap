// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CouponCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ gtVEun"
      >
        <div
          class="sc-gsTCUz kSfVUQ"
        >
          <svg
            fill="none"
            height="36px"
            viewBox="0 0 20 20"
            width="36px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.41 9.58L10.41 0.58C10.05 0.22 9.55 0 9 0H2C0.9 0 0 0.9 0 2V9C0 9.55 0.22 10.05 0.59 10.42L9.59 19.42C9.95 19.78 10.45 20 11 20C11.55 20 12.05 19.78 12.41 19.41L19.41 12.41C19.78 12.05 20 11.55 20 11C20 10.45 19.77 9.94 19.41 9.58ZM3.5 5C2.67 5 2 4.33 2 3.5C2 2.67 2.67 2 3.5 2C4.33 2 5 2.67 5 3.5C5 4.33 4.33 5 3.5 5ZM15.27 13.27L11 17.54L6.73 13.27C6.28 12.81 6 12.19 6 11.5C6 10.12 7.12 9 8.5 9C9.19 9 9.82 9.28 10.27 9.74L11 10.46L11.73 9.73C12.18 9.28 12.81 9 13.5 9C14.88 9 16 10.12 16 11.5C16 12.19 15.72 12.82 15.27 13.27Z"
              fill="#4D4771"
            />
          </svg>
        </div>
        <div
          class="sc-dlfnbm cIVZKb"
        >
          <div
            class="sc-hKgILt hhHlE"
          >
            <h4
              class="sc-gKsewC jzRcuu"
            />
          </div>
        </div>
        <div
          class="sc-jSgupP eplsmV"
        >
          <svg
            fill="none"
            height="20px"
            stroke="#4D4771"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="15"
              x2="9"
              y1="9"
              y2="15"
            />
            <line
              x1="9"
              x2="15"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ gtVEun"
    >
      <div
        class="sc-gsTCUz kSfVUQ"
      >
        <svg
          fill="none"
          height="36px"
          viewBox="0 0 20 20"
          width="36px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.41 9.58L10.41 0.58C10.05 0.22 9.55 0 9 0H2C0.9 0 0 0.9 0 2V9C0 9.55 0.22 10.05 0.59 10.42L9.59 19.42C9.95 19.78 10.45 20 11 20C11.55 20 12.05 19.78 12.41 19.41L19.41 12.41C19.78 12.05 20 11.55 20 11C20 10.45 19.77 9.94 19.41 9.58ZM3.5 5C2.67 5 2 4.33 2 3.5C2 2.67 2.67 2 3.5 2C4.33 2 5 2.67 5 3.5C5 4.33 4.33 5 3.5 5ZM15.27 13.27L11 17.54L6.73 13.27C6.28 12.81 6 12.19 6 11.5C6 10.12 7.12 9 8.5 9C9.19 9 9.82 9.28 10.27 9.74L11 10.46L11.73 9.73C12.18 9.28 12.81 9 13.5 9C14.88 9 16 10.12 16 11.5C16 12.19 15.72 12.82 15.27 13.27Z"
            fill="#4D4771"
          />
        </svg>
      </div>
      <div
        class="sc-dlfnbm cIVZKb"
      >
        <div
          class="sc-hKgILt hhHlE"
        >
          <h4
            class="sc-gKsewC jzRcuu"
          />
        </div>
      </div>
      <div
        class="sc-jSgupP eplsmV"
      >
        <svg
          fill="none"
          height="20px"
          stroke="#4D4771"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="15"
            x2="9"
            y1="9"
            y2="15"
          />
          <line
            x1="9"
            x2="15"
            y1="9"
            y2="15"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Reverse CouponCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdfBwQ gtVEun"
      >
        <div
          class="sc-gsTCUz kSfVUQ"
        >
          <svg
            fill="none"
            height="36px"
            viewBox="0 0 20 20"
            width="36px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M19.41 9.58L10.41 0.58C10.05 0.22 9.55 0 9 0H2C0.9 0 0 0.9 0 2V9C0 9.55 0.22 10.05 0.59 10.42L9.59 19.42C9.95 19.78 10.45 20 11 20C11.55 20 12.05 19.78 12.41 19.41L19.41 12.41C19.78 12.05 20 11.55 20 11C20 10.45 19.77 9.94 19.41 9.58ZM3.5 5C2.67 5 2 4.33 2 3.5C2 2.67 2.67 2 3.5 2C4.33 2 5 2.67 5 3.5C5 4.33 4.33 5 3.5 5ZM15.27 13.27L11 17.54L6.73 13.27C6.28 12.81 6 12.19 6 11.5C6 10.12 7.12 9 8.5 9C9.19 9 9.82 9.28 10.27 9.74L11 10.46L11.73 9.73C12.18 9.28 12.81 9 13.5 9C14.88 9 16 10.12 16 11.5C16 12.19 15.72 12.82 15.27 13.27Z"
              fill="#4D4771"
            />
          </svg>
        </div>
        <div
          class="sc-dlfnbm cIVZKb"
        >
          <div
            class="sc-hKgILt hhHlE"
          >
            <h4
              class="sc-iBPRYJ jiiDDC"
            />
          </div>
        </div>
        <div
          class="sc-jSgupP eplsmV"
        >
          <svg
            fill="none"
            height="20px"
            stroke="#4D4771"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="10"
            />
            <line
              x1="15"
              x2="9"
              y1="9"
              y2="15"
            />
            <line
              x1="9"
              x2="15"
              y1="9"
              y2="15"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdfBwQ gtVEun"
    >
      <div
        class="sc-gsTCUz kSfVUQ"
      >
        <svg
          fill="none"
          height="36px"
          viewBox="0 0 20 20"
          width="36px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.41 9.58L10.41 0.58C10.05 0.22 9.55 0 9 0H2C0.9 0 0 0.9 0 2V9C0 9.55 0.22 10.05 0.59 10.42L9.59 19.42C9.95 19.78 10.45 20 11 20C11.55 20 12.05 19.78 12.41 19.41L19.41 12.41C19.78 12.05 20 11.55 20 11C20 10.45 19.77 9.94 19.41 9.58ZM3.5 5C2.67 5 2 4.33 2 3.5C2 2.67 2.67 2 3.5 2C4.33 2 5 2.67 5 3.5C5 4.33 4.33 5 3.5 5ZM15.27 13.27L11 17.54L6.73 13.27C6.28 12.81 6 12.19 6 11.5C6 10.12 7.12 9 8.5 9C9.19 9 9.82 9.28 10.27 9.74L11 10.46L11.73 9.73C12.18 9.28 12.81 9 13.5 9C14.88 9 16 10.12 16 11.5C16 12.19 15.72 12.82 15.27 13.27Z"
            fill="#4D4771"
          />
        </svg>
      </div>
      <div
        class="sc-dlfnbm cIVZKb"
      >
        <div
          class="sc-hKgILt hhHlE"
        >
          <h4
            class="sc-iBPRYJ jiiDDC"
          />
        </div>
      </div>
      <div
        class="sc-jSgupP eplsmV"
      >
        <svg
          fill="none"
          height="20px"
          stroke="#4D4771"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="20px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="10"
          />
          <line
            x1="15"
            x2="9"
            y1="9"
            y2="15"
          />
          <line
            x1="9"
            x2="15"
            y1="9"
            y2="15"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
